File: shared/clang-wrapper/src/main/java/io/ygdrasil/koreos/clang/ClangInitializationException.java
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang;

/**
 * Exception thrown when libclang initialization fails.
 */
public class ClangInitializationException extends ClangException {
    public ClangInitializationException(String message) {
        super(message);
    }

    public ClangInitializationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: shared/clang-wrapper/src/main/kotlin/io/ygdrasil/koreos/clang/ClangInitializationException.kt
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang

/**
 * Exception thrown when libclang initialization fails.
 */
class ClangInitializationException @JvmOverloads constructor(
    message: String,
    cause: Throwable? = null
) : ClangException(message, cause)
